File: sdk/src/main/java/com/ktx/lib/widget/dialog/XDialog.java
package com.ktx.lib.widget.dialog;

import android.app.Dialog;
import android.content.Context;
import android.view.Gravity;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;


import com.ktx.lib.R;

import java.util.Objects;


/**
 * 通用dialog
 * author: 星辰CentOS
 * on 2019/12/05
 */

public class XDialog extends Dialog {

    private DialogManager mAlert;

    XDialog(@NonNull Context context, int themeResId) {
        super(context, themeResId);
        mAlert = new DialogManager(this, getWindow());
    }

    public void setText(int resId, CharSequence text) {
        mAlert.setText(resId, text);
    }

    public void setOnClickListener(int resId, View.OnClickListener listener) {
        mAlert.setOnClickListener(resId, listener);
    }

    public <T extends View> T getView(int resId) {
        return mAlert.getView(resId);
    }

    public static class Builder {

        private final DialogManager.AlertParams P;

        public Builder(Context context) {
            this(context, R.style.dialog);
        }

        Builder(Context context, int themeResId) {
            P = new DialogManager.AlertParams(context, themeResId);
        }

        public Builder setLayoutView(View view) {
            P.mView = view;
            P.mViewLayoutResId = 0;
            return this;
        }

        public Builder setLayoutView(int layoutResId) {
            P.mView = null;
            P.mViewLayoutResId = layoutResId;
            return this;
        }

        public Builder setCancelable(boolean cancelable) {
            P.mCancelable = cancelable;
            return this;
        }

        public Builder setOnCancelListener(OnCancelListener onCancelListener) {
            P.mOnCancelListener = onCancelListener;
            return this;
        }

        public Builder setOnDismissListener(OnDismissListener onDismissListener) {
            P.mOnDismissListener = onDismissListener;
            return this;
        }

        public Builder setOnKeyListener(OnKeyListener onKeyListener) {
            P.mOnKeyListener = onKeyListener;
            return this;
        }

        public Builder setText(int viewId, CharSequence text) {
            P.mViewMap.put(viewId, text);
            return this;
        }

        public Builder setOnClickListener(int viewId, View.OnClickListener listener) {
            P.mListenerMap.put(viewId, listener);
            return this;
        }

        public Builder fullScreen() {
            P.mWidth = ViewGroup.LayoutParams.MATCH_PARENT;
            return this;
        }

        public Builder fromBottom(boolean isAnimation) {
            if (isAnimation) {
                P.mAnimation = R.style.dialog_from_bottom_anim;
            }
            P.mGravity = Gravity.BOTTOM;
            return this;
        }

        public Builder setWidthAndHeight(int width, int height) {
            P.mWidth = width;
            P.mHeight = height;
            return this;
        }

        public Builder defaultAnimation() {
            P.mAnimation = R.style.dialog_scale_anim;
            return this;
        }

        public Builder setAnimation(int styleAnimation) {
            P.mAnimation = styleAnimation;
            return this;
        }

        public XDialog create() {
            final XDialog dialog = new XDialog(P.mContext, P.mThemeResId);
            P.bind(dialog.mAlert);
            dialog.setCancelable(P.mCancelable);
            dialog.setCanceledOnTouchOutside(P.mCancelable);
            dialog.setOnCancelListener(P.mOnCancelListener);
            dialog.setOnDismissListener(P.mOnDismissListener);
            if (P.mOnKeyListener != null) {
                dialog.setOnKeyListener(P.mOnKeyListener);
            }
            return dialog;
        }

        public XDialog show() {
            final XDialog dialog = create();
            dialog.show();

            return dialog;
        }

        public XDialog transparent() {
            final XDialog dialog = create();
            dialog.show();
            Objects.requireNonNull(dialog.getWindow()).setBackgroundDrawableResource(android.R.color.transparent);
            return dialog;
        }
    }
}

File: lib_common/src/main/java/com/lib/common/dialog/XDialog.kt
package com.lib.common.dialog

import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.view.Gravity
import android.view.View
import android.view.ViewGroup
import com.example.lib_common.R

class XDialog(context: Context, themeResId: Int) : Dialog(context, themeResId) {

    private lateinit var mAlert: DialogManager

    init {
        mAlert = DialogManager(this, window)
    }

    fun setText(resId: Int, text: CharSequence?) {
        mAlert.setText(resId, text)
    }

    fun setOnClickListener(resId: Int, listener: View.OnClickListener?) {
        mAlert.setOnClickListener(resId, listener)
    }

    fun <T : View?> getView(resId: Int): T {
        return mAlert.getView(resId)
    }

    class Builder internal constructor(context: Context, themeResId: Int) {

        private val P: DialogManager.AlertParams

        constructor(context: Context) : this(context, R.style.dialog)

        init {
            P = DialogManager.AlertParams(context, themeResId)
        }

        fun setLayoutView(view: View?): Builder {
            P.mView = view
            P.mViewLayoutResId = 0
            return this
        }

        fun setLayoutView(layoutResId: Int): Builder {
            P.mView = null
            P.mViewLayoutResId = layoutResId
            return this
        }

        fun setCancel(cancelable: Boolean): Builder {
            P.mCancelable = cancelable
            return this
        }

        fun setOnCancelListener(onCancelListener: DialogInterface.OnCancelListener?): Builder {
            P.mOnCancelListener = onCancelListener
            return this
        }

        fun setOnDismissListener(onDismissListener: DialogInterface.OnDismissListener?): Builder {
            P.mOnDismissListener = onDismissListener
            return this
        }

        fun setOnKeyListener(onKeyListener: DialogInterface.OnKeyListener?): Builder {
            P.mOnKeyListener = onKeyListener
            return this
        }

        fun setText(viewId: Int, text: CharSequence?): Builder {
            P.mViewMap.put(viewId, text)
            return this
        }

        fun setOnClickListener(viewId: Int, listener: View.OnClickListener?): Builder {
            P.mListenerMap.put(viewId, listener)
            return this
        }

        fun full(): Builder {
            P.mWidth = ViewGroup.LayoutParams.MATCH_PARENT
            return this
        }

        fun bottom(isAnimation: Boolean): Builder {
            if (isAnimation) {
                P.mAnimation = R.style.dialog_from_bottom_anim
            }
            P.mGravity = Gravity.BOTTOM
            return this
        }

        fun setWidthAndHeight(width: Int, height: Int): Builder {
            P.mWidth = width
            P.mHeight = height
            return this
        }

        fun setAnimation(styleAnimation: Int): Builder {
            P.mAnimation = styleAnimation
            return this
        }

        fun create(): XDialog {
            val dialog = XDialog(P.mContext, P.mThemeResId)
            P.bind(dialog.mAlert)
            dialog.setCancelable(P.mCancelable)
            dialog.setCanceledOnTouchOutside(P.mCancelable)
            dialog.setOnCancelListener(P.mOnCancelListener)
            dialog.setOnDismissListener(P.mOnDismissListener)
            if (P.mOnKeyListener != null) {
                dialog.setOnKeyListener(P.mOnKeyListener)
            }
            return dialog
        }
    }
}